rules/falco_rules.yaml

```diff
@@ -1040,11 +1040,13 @@
               and fd.name startswith "/etc/dd-agent")
 
 - macro: rancher_writing_conf
-  condition: (container.image.repository in (rancher_images)
-              and proc.name in (lb-controller,rancher-dns,healthcheck,rancher-metadat)
-              and (fd.name startswith "/etc/haproxy" or
-                   fd.name startswith "/etc/rancher-dns")
-             )
+  condition: (((proc.name=healthcheck and container.image.repository contains "rancher/healthcheck") or
+               (proc.name=lb-controller and container.image.repository contains "rancher/lb-service-haproxy") or
+               (proc.name=rancher-dns and container.image.repository contains "rancher/dns")) and 
+              (fd.name startswith "/etc/haproxy" or fd.name startswith "/etc/rancher-dns"))
+
+- macro: rancher_writing_root
+  condition: (proc.name=rancher-metadat and container.image.repository contains "rancher/metadata" and fd.name startswith "/answers.json")
```

Comment:
the process name here seems wrong

Reply:
That's because process names that falco sees are truncated at 16 characters. It's a kernel limit: http://lxr.linux.no/linux+v2.6.37/include/linux/sched.h#L245

Reply:
Thanks for the clarification Mark. I have concerns about this but I'll open a separate issue once I have more info.